integration/spark/src/main/scala/org/carbondata/spark/rdd/CarbonGlobalDictionaryRDD.scala

```diff
@@ -321,3 +327,108 @@ class CarbonGlobalDictionaryGenerateRDD(
     iter
   }
 }
+
+/**
+ *  Set column dictionry patition format
+  *
+  * @param id  partition id
+ * @param dimension  current carbon dimension
+ */
+class CarbonColumnDictPatition(id: Int, dimension: CarbonDimension)
+  extends Partition {
+  override val index: Int = id
+  val preDefDictDimension = dimension
+}
+
+/**
+ * Use external column dict to generate global dictionary
+  *
+  * @param carbonLoadModel  carbon load model
+ * @param sparkContext  spark context
+ * @param table  carbon table identifier
+ * @param dimensions  carbon dimenisons having predefined dict
+ * @param hdfsLocation  carbon base store path
+ * @param dictFolderPath  path of dictionary folder
+ */
+class CarbonColumnDictGenerateRDD(carbonLoadModel: CarbonLoadModel,
+       sparkContext: SparkContext,
+       table: CarbonTableIdentifier,
+       dimensions: Array[CarbonDimension],
+       hdfsLocation: String,
+       dictFolderPath: String)
+  extends RDD[(Int, ColumnDistinctValues)](sparkContext, Nil) with Logging {
+
+  override def getPartitions: Array[Partition] = {
+    val primDimensionsBuffer = new ArrayBuffer[CarbonDimension]
+    for (dimension <- dimensions) {
+      val dims = getPrimDimensionWithDict(carbonLoadModel, dimension, true)
+      primDimensionsBuffer ++= dims
+    }
+    val primDimensions = primDimensionsBuffer.toArray
+    val primDimLength = primDimensions.length
+    val result = new Array[Partition](primDimLength)
+    var primColStarIndex = 0
+    for (i <- 0 until primDimLength) {
+      result(i) = new CarbonColumnDictPatition(i, primDimensions(i))
```

Comment:
no need to define new variable "primDimensions"
please directly use variable "primDimensionsBuffer"